src/topics/scheduled.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const winston = require('winston');
+const { CronJob } = require('cron');
+
+const db = require('../database');
+const categories = require('../categories');
+const posts = require('../posts');
+const socketHelpers = require('../socket.io/helpers');
+const topics = require('./index');
+const user = require('../user');
+
+const Scheduled = module.exports;
+
+Scheduled.startJobs = function () {
+	winston.verbose('[scheduled topics] Starting jobs.');
+	new CronJob('*/1 * * * *', Scheduled.handleExpired, null, true);
+};
+
+Scheduled.handleExpired = async function () {
+	const now = Date.now();
+	const tids = await db.getSortedSetRangeByScore('topics:scheduled', 0, -1, '-inf', now);
+
+	if (!tids.length) {
+		return;
+	}
+
+	let topicsData = await topics.getTopicsData(tids);
+	// Filter deleted
+	topicsData = topicsData.filter(topicData => Boolean(topicData));
+	const uids = topicsData
+		.map(topicData => topicData.uid)
+		// Filter guests topics
+		.filter(uid => uid)
+		.filter((uid, idx, arr) => arr.indexOf(uid) === idx);
```

Comment:
_.uniq is much better for getting unique values.

Reply:
Done.